### ISSUE TITLE
EXPERIMENTAL=1 produces failing test_hdf5_impl

### ISSUE THREAD

**Issue by psteinb** [sqeazy/sqeazy#31]
```
$ ./tests/test_hdf5_impl 
Running 23 test cases...
Done registering sqy as hdf5 filter.
Done registering sqy as hdf5 filter.
Done registering sqy as hdf5 filter.
[sqeazy::h5_file, /home/steinbac/development/sqeazy/src/cpp/src/hdf5_utils.hpp:411]     requested dataset IntArray_foo not found in /home/steinbac/development/sqeazy/src/cpp/exp_build/tests/sample.h5
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
terminate called without an active exception
unknown location(0): fatal error: in "hdf5_utils/write_dataset_with_filter": signal: SIGABRT (application abort requested)
/home/steinbac/development/sqeazy/src/cpp/tests/test_hdf5_impl.cpp(347): last checkpoint
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
terminate called recursively
unknown location(0): fatal error: in "hdf5_utils/read_write_dataset_with_filter": signal: SIGABRT (application abort requested)
/home/steinbac/development/sqeazy/src/cpp/tests/test_hdf5_impl.cpp(371): last checkpoint
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
terminate called recursively
unknown location(0): fatal error: in "hdf5_utils/write_compressed_dataset": signal: SIGABRT (application abort requested)
/home/steinbac/development/sqeazy/src/cpp/tests/test_hdf5_impl.cpp(398): last checkpoint: "write_compressed_dataset" entry.
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
terminate called recursively
unknown location(0): fatal error: in "hdf5_utils/roundtrip_compressed_dataset": signal: SIGABRT (application abort requested)
/home/steinbac/development/sqeazy/src/cpp/tests/test_hdf5_impl.cpp(448): last checkpoint: "roundtrip_compressed_dataset" entry.
[sse_bitplane_reorder_encode] 432 (16 bits per element) cannot be processed by 128 without rest
unknown location(0): fatal error: in "hdf5_utils/roundtrip_bswap1_lz4_as_reference": std::runtime_error: /home/steinbac/development/sqeazy/src/cpp/src/dynamic_stage.hpp:438  encode wrote past the end of temporary buffers

/home/steinbac/development/sqeazy/src/cpp/tests/test_hdf5_impl.cpp(536): last checkpoint: "roundtrip_bswap1_lz4_as_reference" entry.

*** 5 failures are detected in the test module "TEST_HDF5_IMPLEMENTATION"
Segmentation fault (core dumped)
```

**Comment by psteinb** [comment]
reproduced in debug mode

**Comment by psteinb** [comment]
fixed in 5e4add5
